src/Compilers/CSharp/Portable/Binder/InMethodBinder.cs

```diff
@@ -158,7 +158,7 @@ internal override GeneratedLabelSymbol ContinueLabel
 
         internal override TypeSymbol GetIteratorElementType(YieldStatementSyntax node, DiagnosticBag diagnostics)
         {
-            TypeSymbol returnType = _methodSymbol.ReturnType;
+            TypeSymbol returnType = (_methodSymbol as LocalFunctionSymbol)?.ReturnTypeIterator ?? _methodSymbol.ReturnType;
```

Comment:
What is the behavior of `ReturnTypeIterator` in the case a local function isn't an iterator.  Is it `null`?